bots/machine/testvm.py

```diff
@@ -18,1407 +18,26 @@
-
-        disk = {
-            "path": image,
-            "serial": serial,
-            "filename": image,
-            "dev": dev,
-            "index": index,
-            "type": type,
-        }
-
-        self._disks.append(disk)
-        return disk
-
-    def rem_disk(self, disk, quick=False):
-        if not quick:
-            disk_desc = TEST_DISK_XML % {
-                'file': disk["filename"],
-                'serial': disk["serial"],
-                'unit': disk["index"],
-                'dev': disk["dev"],
-                'type': disk["type"]
-            }
-
-            if self._domain:
-                if self._domain.detachDeviceFlags(disk_desc, libvirt.VIR_DOMAIN_AFFECT_LIVE ) != 0:
-                    raise Failure("Unable to remove disk from vm")
-
-    def _qemu_monitor(self, command):
-        self.message("& " + command)
-        # you can run commands manually using virsh:
-        # virsh -c qemu:///session qemu-monitor-command [domain name/id] --hmp [command]
-        output = libvirt_qemu.qemuMonitorCommand(self._domain, command, libvirt_qemu.VIR_DOMAIN_QEMU_MONITOR_COMMAND_HMP)
-        self.message(output.strip())
-        return output
-
-    def add_netiface(self, networking=None, vlan=0):
-        if not networking:
-            networking = VirtNetwork().interface()
-        cmd = "device_add virtio-net-pci,mac={0}".format(networking["mac"])
-        if vlan == 0:
-            self._qemu_monitor("netdev_add socket,mcast=230.0.0.1:{mcast},id={id}".format(mcast=networking["mcast"], id=networking["hostnet"]))
-            cmd += ",netdev={id}".format(id=networking["hostnet"])
-        else:
-            cmd += ",vlan={vlan}".format(vlan=vlan)
-        self._qemu_monitor(cmd)
-        return networking["mac"]
+import sys
+import os
 
-    def needs_writable_usr(self):
-        # On atomic systems, we need a hack to change files in /usr/lib/systemd
-        if self.atomic_image:
-            self.execute(command="mount -o remount,rw /usr")
+# ensure that this path is there in case of symlinks to testvm.py
+core_dir = os.path.dirname(os.path.realpath(__file__))
+if not core_dir in sys.path:
+    sys.path.insert(1, core_dir)
 
-    def wait_for_cockpit_running(self, address="localhost", port=9090, seconds=30):
-        WAIT_COCKPIT_RUNNING = """#!/bin/sh
-until curl -s --connect-timeout 2 --max-time 3 http://%s:%s >/dev/null; do
-    sleep 0.5;
-done;
-""" % (address, port)
-        with Timeout(seconds=seconds, error_message="Timeout while waiting for cockpit to start"):
-            self.execute(script=WAIT_COCKPIT_RUNNING)
+from machine_core.timeout import Timeout
+from machine_core.machine_base import Machine
+from machine_core.exceptions import Failure, RepeatableFailure
+from machine_core.machine_virtual import VirtMachine, VirtNetwork, get_build_image, get_test_image
+from machine_core.constants import BOTS_DIR, TEST_DIR, DEFAULT_IMAGE, TEST_OS_DEFAULT
```

Comment:
These are all unused. I. e. they are for providing a backwards compatible API (which is good and necessary), but pyflakes whines (this is what's making the semaphore test fail):
```
❱❱❱ tools/test-static-code 
1..6
bots/machine/testvm.py:30: 'machine_core.timeout.Timeout' imported but unused
bots/machine/testvm.py:31: 'machine_core.machine_base.Machine' imported but unused
bots/machine/testvm.py:32: 'machine_core.exceptions.Failure' imported but unused
bots/machine/testvm.py:32: 'machine_core.exceptions.RepeatableFailure' imported but unused
bots/machine/testvm.py:33: 'machine_core.machine_virtual.VirtMachine' imported but unused
bots/machine/testvm.py:33: 'machine_core.machine_virtual.VirtNetwork' imported but unused
bots/machine/testvm.py:33: 'machine_core.machine_virtual.get_build_image' imported but unused
bots/machine/testvm.py:33: 'machine_core.machine_virtual.get_test_image' imported but unused
bots/machine/testvm.py:34: 'machine_core.constants.BOTS_DIR' imported but unused
bots/machine/testvm.py:34: 'machine_core.constants.TEST_DIR' imported but unused
bots/machine/testvm.py:34: 'machine_core.constants.DEFAULT_IMAGE' imported but unused
bots/machine/testvm.py:34: 'machine_core.constants.TEST_OS_DEFAULT' imported but unused
not ok 1 pyflakes bots pkg tools
test/avocado/run-tests:34: redefinition of unused 'subprocess' from line 25
```

I suggest doing something like

```py
__all__ = [Timeout, Machine, Failure, ...]
```

which will both point out that these are exported API, and also reference the identifiers so that pyflakes stops complaining.